src/core_plugins/kibana/public/management/sections/indices/create_index_pattern_wizard/create_index_pattern_wizard.js

```diff
@@ -1,307 +1,232 @@
-        getIndices(exactSearchQuery),
-        getIndices(partialSearchQuery),
-        createReasonableWait()
-      ])
-        .then(([
-          matchingIndicesResponse,
-          partialMatchingIndicesResponse
-        ]) => {
-          if (thisFetchMatchingIndicesRequest === mostRecentFetchMatchingIndicesRequest) {
-            matchingIndices = matchingIndicesResponse;
-            partialMatchingIndices = partialMatchingIndicesResponse;
-            updateWhiteListedIndices();
-            this.isFetchingMatchingIndices = false;
-          }
-        }).catch(error => {
-          notify.error(error);
-        });
-    };
-
-    this.fetchExistingIndices = () => {
-      this.isFetchingExistingIndices = true;
-      const allExistingLocalIndicesPattern = '*';
-
-      Promise.all([
-        getIndices(allExistingLocalIndicesPattern),
-        createReasonableWait()
-      ])
-        .then(([allIndicesResponse]) => {
-          // Cache all indices.
-          allIndices = allIndicesResponse;
-          updateWhiteListedIndices();
-          this.isFetchingExistingIndices = false;
-        }).catch(error => {
-          notify.error(error);
-          this.isFetchingExistingIndices = false;
-        });
-    };
-
-    this.isSystemIndicesCheckBoxVisible = () => (
-      this.wizardStep === 'indexPattern'
+    services.indexPatterns.cache.clear(createdId);
+    services.kbnUrl.change(`/management/kibana/indices/${createdId}`);
+  }
+
+  goToTimeFieldStep = (indexPattern) => {
+    this.setState({ step: 2, indexPattern });
+  }
+
+  goToIndexPatternStep = () => {
+    this.setState({ step: 1 });
+  }
+
+  onChangeIncludingSystemIndices = () => {
+    this.setState(state => ({
+      isIncludingSystemIndices: !state.isIncludingSystemIndices,
+    }));
+  }
+
+  renderHeader() {
+    const { isIncludingSystemIndices } = this.state;
+
+    return (
+      <div>
+        <EuiSpacer size="m"/>
+        <EuiTitle>
+          <h1>Create index pattern</h1>
+        </EuiTitle>
+        <EuiFlexGroup justifyContent="spaceBetween" alignItems="flexEnd">
+          <EuiFlexItem grow={false}>
+            <EuiText>
+              <p>
+                <EuiTextColor color="subdued">
+                  Kibana uses index patterns to retrieve data from Elasticsearch indices for things like visualizations.
+                </EuiTextColor>
+              </p>
+            </EuiText>
+          </EuiFlexItem>
+          <EuiFlexItem grow={false}>
+            <EuiSwitch
+              label="Include system indices"
+              checked={isIncludingSystemIndices}
+              onChange={this.onChangeIncludingSystemIndices}
+            />
+          </EuiFlexItem>
+        </EuiFlexGroup>
+        <EuiSpacer size="m"/>
+      </div>
     );
+  }
 
-    this.goToIndexPatternStep = () => {
-      this.wizardStep = 'indexPattern';
-    };
+  renderInitialLoadingState() {
```

Comment:
When the user clicks the "Check for new data" button are we going to set the `isInitiallyLoadingIndices` state variable back to `true` and then use this method to re-render the loading state? If so, then I'm not sure I get the  meaning behind `initial` in this method name and in `isInitiallyLoadingIndices`?